frontend/src/metabase/lib/dom.js

```diff
@@ -26,7 +26,7 @@ window.METABASE = true;
 // used for detecting if we're previewing an embed
 export const IFRAMED_IN_SELF = (function () {
   try {
-    return window.self !== window.top && window.top.METABASE;
+    return window.self !== window.parent && window.parent.METABASE;
```

Comment:
`IFRAMED_IN_SELF` was only used in a few places. And all of them relate to how we're rendering the embed preview.

This was failing in Cypress because there are 3 frames in Cypress
- Cypress
    - Metabase Iframe
        - Embed preview Iframe

But in our application, there are only 2 frames
- Metabase
    - Embed preview Iframe
 
For this particular code, when running in Cypress `window.parent` will get Cypress frame, so the function will return `false`, which was wrong.

Changing it to `window.parent`, makes sure it returns the immediate parent frame which will be Metabase frame in both cases.